doc/kubernetes/container-image-management.md

```diff
@@ -0,0 +1,76 @@
+# Why container image management
+
+After we support container upgrade via k8s, for the container image, we have two problem to be fixed.
+- After we upgrade one container to a higher version via k8s, if k8s control plane is down and worker reboot, container need to go back to local mode, local mode container should run the higher version other than the original version.
+- Upgrade will bring a new version image to device, after upgrade many times, the disk will be full, need to avoid this.
+
+So we need to manage the container image to fix these two problems
+
+# Existing designs need to know first
+### 1. local mode and kube mode
+The container has two modes. Kube mode means that the contianer is deployed by k8s, the container image version is defined in daemonset like below. The image need to be downloaded from ACR, so it's with ACR prefix.
+```
+sonick8scue.azurecr.io/docker-sonic-telemetry:v1
+```
+ Local mode means that the container is created and started by systemd service, and systemd service will always use local latest image like below to create container.
+ ```
+ docker-sonic-telemetry:latest
+ ```
+
+### 2. Kube container will not replace the local container when their image versions are the same.
+##### why this design
+when the kube container and local container are in the same version, local container should be more stable, becasue it has run for some time. And when device first join k8s cluster as worker, k8s will try to deploy a same version container with local, if we choose replace, the container will has a down time, actually no upgrade happens, down time is not necessary.
+##### one note
+The kube container will dry-run, no real services start inside the container. But it still downloads the sonick8scue.azurecr.io/docker-sonic-telemetry:v1 image on device.
+# How to handle the two problems
```

Comment:
let's focus on the specific issue.  1) How to update tag the local contain image 2) How to clean up obsolete container image